tools/pythonpkg/src/native/python_objects.cpp

```diff
@@ -500,9 +471,9 @@ py::object PythonObject::FromValue(const Value &val, const LogicalType &type,
 		int32_t year, month, day;
 		if (!duckdb::Date::IsFinite(date)) {
 			if (date == date_t::infinity()) {
-				return py::reinterpret_borrow<py::object>(import_cache.datetime.date.max());
```

Comment:
After this change fetching a value from a `datetime` column can return a `str` value which seems unintuitive to me. I tested what `psycopg2` (i.e. the Postgres client) does and it seems to return `datetime max`:

```py
>>> c.execute("select 'infinity'::timestamp")
>>> c.fetchall()
[(datetime.datetime(9999, 12, 31, 23, 59, 59, 999999),)]
>>> c.execute("select '-infinity'::timestamp")
>>> c.fetchall()
[(datetime.datetime(1, 1, 1, 0, 0),)]
```

I would vote for changing this only on the ingestion side so that we don't interpret `datetime max` as infinity, but still convert `infinity` to `datetime max` when fetching values. This loses the ability to round-trip but I don't think that is that important in this case.

Reply:
Hmm that's fair, I had that, but then I figured the roundtrip ability would be preferable over returning a `datetime.datetime` object, since we're already dealing with objects here

But I'll just revert the last commit then 👍